### PR TITLE
use glossarist model in term refs #113

### LINKED ISSUE #113: Convert to use Glossarist gem and models
We should use the Glossarist gem and classes instead of plain text.

https://github.com/metanorma/stepmod-utils/blob/ccfa5e9dd0bc0070098fc09f2676735476881ab2/lib/stepmod/utils/converters/term.rb#L8-L44

### PR DESCRIPTION
**Pull request by HassanAkbar**
- Added a model named `Term`
- Used that in `Term Converter`

Resolves #113

### CODE REVIEW

File: lib/stepmod/utils/term.rb [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+      def to_mn_adoc
+        mn_adoc = ["=== #{definition}"]
+        mn_adoc << "\nalt:[#{acronym}]" if acronym
+        mn_adoc << "\n\n#{designations.join(", ")}" if designations&.any?
+
+        mn_adoc.join
+      end
+
+      class << self
+        def from_h(hash)
```
Comment (ronaldtse) [inline lib/stepmod/utils/term.rb:18]: Can we use `Glossarist::LocalizedConcept(definition: "XXX", etc)` instead? Not a fan of additional setting methods.

**Review by ronaldtse — CHANGES_REQUESTED** [review]

File: lib/stepmod/utils/term.rb (hunk 1 above)
Comment (HassanAkbar) [reply lib/stepmod/utils/term.rb:18]: So should we move the code of this method to `initialize` method inside this class like below ? :

```ruby
module Stepmod
  module Utils
    class Term < Glossarist::LocalizedConcept
      # Term acronym
      def initialize
        ...
        Glossarist::LocalizedConcept.from_h(definition: "XXX", etc)
      end
    end
  end
end
```

And call it like `Stepmod::Utils::Term.new(definition: "XXX")` from [here](https://github.com/metanorma/stepmod-utils/pull/118/files/13dccd824f74717a6bed89a589c282ccbe5ac363#diff-7f6186dcdbfd8cb4fbf8df3809ba8e6abbe34a46b3e0c4103d8bc3200cf78539R37).

Comment (ronaldtse) [reply lib/stepmod/utils/term.rb:18]: @HassanAkbar LocalizedConcept already has an initialize method that accepts a hash and the attributes that we need:
https://github.com/glossarist/glossarist-ruby/blob/d6a746e22ed71665bfadf345dac1d3d70b170a8d/lib/glossarist/localized_concept.rb#L100-L104

I don't think we need a new method here.

Comment (HassanAkbar) [reply lib/stepmod/utils/term.rb:18]: The `acronym` is missing in `LocalizedConcept` and currently acronyms and synonyms are shown in different lines in the output. We can do either one of the following:

- We can combine both (acronyms and synonyms) and use `designations` to store them together and show them in a single line.
- We can update `glossarist` gem and add a attribute for `acronym` in `LocalizedConcept`. Currently we are using [glossarist version 0.1.0](https://github.com/glossarist/glossarist-ruby/tree/v0.1.0).

Comment (ronaldtse) [reply lib/stepmod/utils/term.rb:18]: We should update the Glossarist gem. In this case let's merge this first and update Glossarist.

Comment (ronaldtse) [reply lib/stepmod/utils/term.rb:18]: This will be handled here: https://github.com/glossarist/glossarist-ruby/issues/26

**Pull request merged**